vllm/core/scheduler.py

```diff
@@ -395,6 +395,38 @@ def __init__(
         # for processing and deallocation by the free_finished_seq_groups()
         self._async_stopped: List[SequenceGroup] = []
 
+        # For chunked prefill, we allocate a set of "prefill slots" that
+        # each represent a sequence group that can be partially prefilled.
+        # Having multiple partial prefills in flight allows us to minimize TTFT
+        # and avoid decode starvation in cases where a single sequence group
+        # with a very large prompt blocks the queue for too many iterations.
+        self.num_prefill_slots = scheduler_config.num_prefill_slots
+        self.prefill_slots_running = 0
+        self.big_prefill_requests = 0
+        # Requests with more than (4% max context length) tokens to prefill
+        # are "big".
+        # The number of big prefill requests is limited so that smaller
+        # requests may jump the queue in front of them and get to the decode
+        # phase faster.
+        self.big_prefill_threshold = scheduler_config.max_model_len // 25
+        self.max_big_requests = 1  # TODO: something
```

Comment:
We think that leaving this set to `1` is reasonable, since in the worst case when there are many large prompt requests this is identical to the existing behavior. We could add a new cli arg or environment variable to adjust it, but we also don't really want to pollute the config surface even more.

Reply:
Well they could always turn off the feature entirely :D

But yeah you're right. Do you mind if this is a` vllm.envs` setting instead of another cli arg?